examples/training/classification/densenet/imagenet/basic_training.py

```diff
@@ -0,0 +1,230 @@
+    validation_dataset = tf.data.TFRecordDataset(filenames=validation_filenames)
+
+    train_dataset = train_dataset.map(
+        parse_imagenet_example,
+        num_parallel_calls=tf.data.AUTOTUNE,
+    )
+    validation_dataset = validation_dataset.map(
+        parse_imagenet_example,
+        num_parallel_calls=tf.data.AUTOTUNE,
+    )
+
+    return train_dataset.batch(BATCH_SIZE), validation_dataset.batch(BATCH_SIZE)
+
+
+train_ds, test_ds = load_imagenet_dataset()
+
+
+"""
+Next, we augment our dataset. We define a set of augmentation layers and then apply them to our input dataset using the `apply_augmentation` method from our KerasCV training utils.
+"""
+
+AUGMENT_LAYERS = [
+    keras_cv.layers.RandomFlip(),
+    keras_cv.layers.RandAugment(value_range=(0, 255), magnitude=0.3),
+    keras_cv.layers.RandomCutout(height_factor=0.1, width_factor=0.1),
+]
+
+
+@tf.function
+def augment(img, label):
+    inputs = {"images": img, "labels": label}
+    for layer in AUGMENT_LAYERS:
+        inputs = layer(inputs)
+    return inputs["images"], inputs["labels"]
+
+
+train_ds = train_ds.map(augment, num_parallel_calls=tf.data.AUTOTUNE).prefetch(
+    tf.data.AUTOTUNE
+)
+test_ds = test_ds.prefetch(tf.data.AUTOTUNE)
+
+
+"""
+Now we can begin training our model. We begin by loading a DenseNet model from KerasCV.
+"""
+
+
+def get_model():
+    return DenseNet121(
+        include_rescaling=True,
+        include_top=True,
+        num_classes=NUM_CLASSES,
+        input_shape=IMAGE_SIZE + (3,),
+    )
+
+
+"""
+Next, we pick an optimizer. Here we use Adam with a linearly decaying learning rate.
+"""
+
+
+def get_optimizer():
+    return optimizers.Adam(
+        learning_rate=optimizers.schedules.PolynomialDecay(
+            initial_learning_rate=0.005,
+            decay_steps=train_ds.cardinality().numpy() * EPOCHS / 2,
+            end_learning_rate=0.0001,
+        )
+    )
+
+
+"""
+Next, we pick a loss function. Here we use a built-in Keras loss function, so we simply specify it as a string.
+"""
+
+
+def get_loss_fn():
+    return losses.CategoricalCrossentropy(label_smoothing=0.1)
+
+
+"""
+Next, we specify the metrics that we want to track. For this example, we track accuracy. Once again, accuracy is a built-in metric in Keras so we can specify it as a string.
+"""
+
+
+def get_metrics():
+    return ["accuracy"]
+
+
+"""
+As a last piece of configuration, we configure callbacks for the method. We use EarlyStopping, BackupAndRestore, and a model checkpointing callback.
+"""
+
+
+def get_callbacks():
+    return [
+        callbacks.EarlyStopping(patience=30),
+        callbacks.BackupAndRestore(FLAGS.backup_path),
+        callbacks.ModelCheckpoint(FLAGS.weights_path),
+        callbacks.TensorBoard(log_dir=FLAGS.tensorboard_path),
+    ]
+
+
+"""
+We can now compile the model and fit it to the training dataset.
+"""
+
+with tf.distribute.MirroredStrategy().scope():
```

Comment:
Nit: we should be able to JUST construct the model under scope and unindent the rest.

```
with strategy.scope():
    model = get_model()
  
model.compile()

model.fit(train_ds, ...)
```

@rchao to confirm this.

Reply:
Correct, we should expect this to work. Please let me know if not.